integreat_cms/static/src/js/pois/poi-map.ts

```diff
@@ -1,165 +1,242 @@
-import maplibregl from "maplibre-gl";
+import { FullscreenControl, LngLat, LngLatBoundsLike, Map, MapOptions, MapMouseEvent, Marker } from "maplibre-gl";
 import { getCsrfToken } from "../utils/csrf-token";
 import { updateField } from "./poi-actions";
 
 window.addEventListener("load", () => {
-    let longitude = document.getElementById("id_longitude") as HTMLInputElement;
-    let latitude = document.getElementById("id_latitude") as HTMLInputElement;
+    // Render the map initially
+    const [map, marker] = renderMap() as [Map, Marker];
 
-    showPreview();
-    longitude.addEventListener("focusout", showPreview);
-    latitude.addEventListener("focusout", showPreview);
+    // Update the map every time a coordinate changes
+    const [latitudeInput, longitudeInput] = getCoordinateInputFields();
+    latitudeInput.addEventListener("focusout", updateMap.bind(null, map, marker));
+    longitudeInput.addEventListener("focusout", updateMap.bind(null, map, marker));
```

Comment:
I think it doesn't make sense to render the complete map new each time the coordinates change - just render the map once initially and after that, we can change the position via [`flyTo()`](https://maplibre.org/maplibre-gl-js-docs/api/map/#map#flyto) (which also looks a bit nicer)...